### PR TITLE
Explictly check at startup that we don't support `bitcoind`'s `blocksonly` option.

### PR DESCRIPTION
**Pull request by darosior**
This adds a sanity check function to `bcli` which we could extend with other sanity checks in the future.

This was [brought up](https://github.com/ElementsProject/lightning/issues/3858#issuecomment-665619664) by @ZmnSCPxj.

### CODE REVIEW

**Comment by ZmnSCPxj** [comment]
bfc73eb4f52c61aa22934039f75a777c4c32ee76 in #3870 also does a `getnetworkinfo`, but rather than a separate call, it replaces the `echo` in `wait_for_bitcoind`, what do you think?  Since we need to call `getnetworkinfo` anyway I think replacing the `echo` would be better (though this is arguably a tiny optimization).

**Comment by ZmnSCPxj** [comment]
Also we should probably update `README.md` as well, nobody reads anything in `doc/` anyway.

**Comment by darosior** [comment]
I updated the README in the first commit.

Regarding replacing `echo`, I agree it's cleaner after having read your commit. Will amend!

**Comment by darosior** [comment]
Adapted part of https://github.com/ElementsProject/lightning/commit/bfc73eb4f52c61aa22934039f75a777c4c32ee76 in https://github.com/ElementsProject/lightning/pull/3889/commits/b934a927457dbce46b9ef93dbb0c00f2996b60f9 and built upon it for transaction relay sanity checks. Added a trivial check for `bitcoind` minimal supported version too. Also, this is a bit more restrictive than your initial version, as we fail to start if the call fails. I think it makes sense to not go further if a simple `getnetworkinfo` does not succeed.

File: README.md [hunk 1]
```diff
@@ -31,7 +31,7 @@ Don't hesitate to reach out to us on IRC at [#lightning-dev @ freenode.net][irc1
 
 ## Getting Started
 
-c-lightning only works on Linux and Mac OS, and requires a locally (or remotely) running `bitcoind` (version 0.16 or above) that is fully caught up with the network you're testing on.
+c-lightning only works on Linux and Mac OS, and requires a locally (or remotely) running `bitcoind` (version 0.16 or above) that is fully caught up with the network you're runing on, and relays transactions (ie with `blocksonly=0`).
```
Comment (ZmnSCPxj) [inline README.md:34]: s/runing/running/

File: contrib/pyln-testing/pyln/testing/utils.py [hunk 2]
```diff
@@ -1043,7 +1043,8 @@ def get_node(self, node_id=None, options=None, dbfile=None,
         db = self.db_provider.get_db(os.path.join(lightning_dir, TEST_NETWORK), self.testname, node_id)
         node = self.node_cls(
             node_id, lightning_dir, self.bitcoind, self.executor, db=db,
-            port=port, options=options, **kwargs
+            port=port, options=options, may_fail=may_fail or expect_fail,
```
Comment (ZmnSCPxj) [inline contrib/pyln-testing/pyln/testing/utils.py:1046]: What we usually do in existing tests is `may_fail=True, expect_fail=True` but I agree this is better and makes sense.

Comment (darosior) [reply contrib/pyln-testing/pyln/testing/utils.py:1046]: Yeah, and `LightningNode()` does not have an `expect_fail` parameter either.

**Comment by darosior** [comment]
Corrected the grammar error.

File: plugins/bcli.c [hunk 3]
```diff
@@ -834,14 +837,62 @@ static void bitcoind_failure(struct plugin *p, const char *error_message)
...
+	 * estimates. */
+	relaytok = json_get_member(buf, result, "localrelay");
+	if (!relaytok || !json_to_bool(buf, relaytok, &tx_relay))
+		plugin_err(p, "No 'localrelay' in '%s' ? Got '%s'. Can not"
+			      " continue without proceeding to sanity checks.",
+			      gather_args(bitcoind, "getnetworkinfo", NULL), buf);
+
+	if (!tx_relay)
+		plugin_err(p, "The 'blocksonly' mode of bitcoind, or any option "
+			      "deactivating transaction relay is not supported.");
```
Comment (niftynei) [inline plugins/bcli.c:880]: Maybe offer a suggestion of how to fix it? i.e. 

```suggestion
			      "deactivating transaction relay is not supported. Remove -blocksonly from your bitcoind config");
```

File: plugins/bcli.c [hunk 4]
```diff
@@ -834,14 +837,62 @@ static void bitcoind_failure(struct plugin *p, const char *error_message)
...
+static void parse_getnetworkinfo_result(struct plugin *p, const char *buf)
+{
+	const jsmntok_t *result, *versiontok, *relaytok;
+	bool valid, tx_relay;
+
+	result = json_parse_input(NULL,
+				  buf, strlen(buf),
+				  &valid);
+	if (!result || !valid)
+		plugin_err(p, "No or invalid response to '%s' ? Got '%s'. Can not"
```
[the remark this answers is not in the document]
Comment (darosior) [reply plugins/bcli.c:850]: Done

File: plugins/bcli.c [hunk 5]
```diff
@@ -834,14 +837,62 @@ static void bitcoind_failure(struct plugin *p, const char *error_message)
...
+			      " continue without proceeding to sanity checks.",
+			      gather_args(bitcoind, "getnetworkinfo", NULL), buf);
+
+	if (!json_to_u32(buf, versiontok, &bitcoind->version))
+		plugin_err(p, "Invalid 'version' in '%s' ? Got '%s'. Can not"
+			      " continue without proceeding to sanity checks.",
+			      gather_args(bitcoind, "getnetworkinfo", NULL), buf);
+
+	if (bitcoind->version < 160000)
+		plugin_err(p, "Unsupported bitcoind version, you need to update"
```
[the remark this answers is not in the document]
Comment (darosior) [reply plugins/bcli.c:867]: Done

File: plugins/bcli.c (hunk 3 above)
Comment (darosior) [reply plugins/bcli.c:880]: Seems a bit redundant ?..
> The 'blocksonly' mode of bitcoind [...] is not supported. Remove -blocksonly from your bitcoind config

Comment (ZmnSCPxj) [reply plugins/bcli.c:881]: Agree as well.

**Pull request merged**